File: sdks/csharp/src/NovaBilling/Types/NetRevenueResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record NetRevenueResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("grossRevenue")]
    public required double GrossRevenue { get; set; }

    [JsonPropertyName("refunds")]
    public required double Refunds { get; set; }

    [JsonPropertyName("creditNotes")]
    public required double CreditNotes { get; set; }

    [JsonPropertyName("netRevenue")]
    public required double NetRevenue { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/net_revenue_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class NetRevenueResponse < Internal::Types::Model
      field :gross_revenue, -> { Integer }, optional: false, nullable: false, api_name: "grossRevenue"
      field :refunds, -> { Integer }, optional: false, nullable: false
      field :credit_notes, -> { Integer }, optional: false, nullable: false, api_name: "creditNotes"
      field :net_revenue, -> { Integer }, optional: false, nullable: false, api_name: "netRevenue"
    end
  end
end
